### PR TITLE
feat(#110): add post-fix targeted prosecution pass to Code-Conductor workflow

### LINKED ISSUE #110: Code-Conductor: add post-fix targeted prosecution pass after Code-Smith applies Critical/High findings
## Problem

The adversarial review pipeline is **one-shot**: prosecution → defense → judge runs on the pre-fix code. After the judge categorizes findings and Code-Conductor routes accepted fixes to Code-Smith, the **post-fix code is never re-reviewed adversarially**. The Validation Ladder (Tiers 1–3) catches build/lint/test regressions, but structural logic bugs introduced by the fix itself fall through.

**Origin incident**: PR #110 in Countdown-Clash — a D1 fix moved `resolvedBalls.delete` outside a conditional but retained it inside a for-loop, introducing a within-loop guard-clearing bug. Three prosecution passes reviewed the pre-fix code; no adversarial lens ever saw the post-fix code. Gemini's external review caught it; the internal pipeline did not.

### Post-Fix Targeted Prosecution Step

A **full adversarial review cycle** (prosecution → defense → judge) runs after Code-Smith applies accepted fixes, tightly scoped to only the fix diff.

### Key Decisions

| Decision | Choice | Rationale |
|----------|--------|-----------|
| Pipeline depth | Full (prosecution → defense → judge) | Maximum rigor; tight scope keeps it proportionate |
| Prosecution passes | 3 parallel | Consistent with main review; origin incident proved small diffs hide subtle bugs; coverage variance matters even on narrow scope |
| Trigger | Critical/High fix, OR any fix modifying control flow (guards, loops, conditionals) | Catches highest-risk cases + the specific origin-incident bug class |
| Loop budget | 1 | Fix once, then escalate to user. Prevents recursive cost explosion |
| Scope | Fix diff only | Code-Conductor computes fix-diff files via `git diff` and passes only those files + diff hunks in the prosecution prompt. Code-Critic operates in normal code-prosecution mode with a constrained input — no new mode needed |
| Recursion guard | First-class constraint | Post-fix prosecution never triggers another post-fix prosecution. Absolute rule, not an edge case |

### Flow

```
Main Review (prosecution → defense → judge)
  → Code-Conductor routes accepted fixes to specialists
  → Code-Smith / Test-Writer / Doc-Keeper apply all accepted fixes
  → Trigger check: ≥1 Critical/High fix, OR any fix modifies control flow?
    → YES: Post-fix targeted prosecution (3 passes, diff-scoped)
      → Diff-scoping: Code-Conductor runs `git diff` on the applied fixes,
        passes only changed files + hunks in each prosecution prompt
      → Defense pass on merged post-fix findings ledger
      → Judge pass (Code-Review-Response)
      → If new issues accepted:
        → Route to specialists (loop budget: 1)
        → NO second post-fix prosecution (recursion guard)
        → If fixes fail validation: escalate to user via askQuestions
      → If clean: proceed to CE Gate / PR
    → NO: proceed to CE Gate / PR (no post-fix review needed)
```

### Integration Points (from adversarial design challenge)

**Interruption Budget**: Post-fix review is a **separate review cycle** with its own interruption budget. Escalation (loop budget exhausted) is a legitimate interruption regardless of main-review budget consumption.

**Pipeline Metrics**: Post-fix findings get separate fields in the metrics block:
- `postfix_prosecution_findings`, `postfix_defense_disproved`, `postfix_judge_accepted`, `postfix_rework_cycles`

**Adversarial Review Scores table**: Add a **"Post-fix Review"** row to the PR body scores table (separate from the main "Code Review" row).

**Validation Ladder Tier 4**: Update description to reflect the post-fix step between fix routing and CE Gate:
- Tier 4: Independent review → **fix routing → post-fix targeted prosecution (if triggered)** → CE Gate

**AC Cross-Check Gate interaction**: If a post-fix finding outside the fix diff relates to an explicit acceptance criterion, the AC Cross-Check Gate overrides the "DEFERRED-SIGNIFICANT" default — AC violations cannot be deferred (absolute rule takes precedence).

**CE Gate ordering**: Post-fix review completes **before** CE Gate begins. Sequential flow, not overlapping. The loop budgets are independent: post-fix review (1 cycle) → CE Gate (2 cycles).

### Edge Cases

1. **Findings outside fix diff** → Flag as DEFERRED-SIGNIFICANT (subject to AC Cross-Check Gate override)
2. **Multiple specialist rounds** → Trigger check runs once after ALL accepted fixes are applied, not after each individual specialist call
3. **No accepted findings from main review** → Skip entirely (no fixes = no post-fix review)
4. **Cost acknowledgment** → 5 additional subagent invocations per triggered review (3 prosecution + 1 defense + 1 judge). Accepted tradeoff: coverage on the highest-risk code region (just-modified fix code) justifies the cost

### Testing Scope

Documentation-only change (agent `.md` file). No code to test. Validation via existing quick-validate commands (grep for retired terms, agent count check).

## Acceptance Criteria

- [ ] Code-Conductor.agent.md includes a post-fix targeted prosecution step in the Post-Judgment Fix Routing section
- [ ] Step specifies: 3 prosecution passes, diff-scoped, full pipeline (prosecution → defense → judge)
- [ ] Diff-scoping mechanism documented: Code-Conductor computes fix-diff file list via `git diff`, passes only changed files + hunks in prosecution prompts
- [ ] Trigger condition documented: ≥1 Critical or High finding accepted and fixed, OR any fix that modifies control flow (guards, loops, conditionals)
- [ ] Loop budget of 1 documented: fix once, then escalate via askQuestions
- [ ] Recursion guard as first-class constraint: post-fix prosecution never triggers another post-fix prosecution
- [ ] Step placement: after all specialists apply accepted fixes, before CE Gate / PR creation
- [ ] Interruption budget: post-fix review documented as a separate cycle
- [ ] Pipeline Metrics: separate `postfix_*` fields specified
- [ ] Adversarial Review Scores table: "Post-fix Review" row added
- [ ] Validation Ladder Tier 4: description updated to include post-fix step
- [ ] AC Cross-Check Gate interaction: documented (AC gate overrides DEFERRED-SIGNIFICANT for AC-related findings)

## Rejected Alternatives

- **Single-shot prosecution (no defense/judge)**: Faster but inconsistent with the pipeline's adversarial principle. False positives in post-fix findings would be accepted without challenge.
- **1 prosecution pass**: Lower cost but reduced coverage variance. The origin incident (a subtle within-loop bug in a small fix) is exactly the class of defect that benefits from multiple independent review perspectives.
- **Always trigger (any accepted finding)**: Disproportionate — trivial fixes (typo corrections, comment updates) don't warrant a full 3-pass review cycle.
- **Loop budget of 0 (immediate escalation)**: Too conservative — a simple coding error in the fix shouldn't require user intervention.
- **New Code-Critic mode (delta review)**: Over-engineering — constrained file list via prompt achieves the same scoping without adding a new mode to maintain.

## Adversarial Challenge Summary

Design was stress-tested with Code-Critic (design prosecution). 10 challenges raised:
- 1 critical (diff-scoping mechanism) — incorporated: prompt-scoped via file list
- 2 high (interruption budget, pipeline metrics) — incorporated: separate cycle, separate fields
- 5 medium (scores table, AC gate interaction, CE Gate ordering, Tier 4 description, pass count reconsideration) — 4 incorporated, 1 dismissed (pass count: user decision)
- 2 low (cost quantification, recursion guard placement) — 1 incorporated (recursion guard promoted), 1 dismissed (cost noted)

## References

- Origin incident: Countdown-Clash PR #110, finding GR-1 (Gemini, Critical)
- Code-Conductor Post-Judgment Fix Routing: `.github/agents/Code-Conductor.agent.md`
- Code-Critic modes: `.github/agents/Code-Critic.agent.md`
- Code-Review-Response judge protocol: `.github/agents/Code-Review-Response.agent.md`

### PR DESCRIPTION
**Pull request by Grimblaz**
# Post-Fix Targeted Prosecution Pass

Adds a mandatory post-fix adversarial review pass to Code-Conductor's workflow. After the main review cycle routes accepted fixes to specialists, a full prosecution → defense → judge pipeline is run against the fix diff to catch defects introduced during remediation.

## Summary

**Problem:** Accepted review fixes were applied by specialists with no adversarial re-review. A fix that corrects one issue could introduce new defects, inconsistencies, or control-flow regressions — and these would ship undetected.

**Solution:** A `#### Post-Fix Targeted Prosecution Pass` subsection under `### Post-Judgment Fix Routing` that:
- Triggers when ≥1 Critical/High finding was fixed **or** any fix touches control flow
- Scopes prosecution to the fix diff only (`git diff HEAD -- {specialist-touched files}`)
- Runs the full 3-pass prosecution → defense → judge pipeline
- Loop budget: 1 (remaining issues converge to DEFERRED-SIGNIFICANT with auto-tracking)
- Recursion guard: post-fix prosecution never triggers another post-fix pass (absolute, not an edge case)

## Changed Files

| File | Change |
|------|--------|
| `.github/agents/Code-Conductor.agent.md` | New `#### Post-Fix Targeted Prosecution Pass` subsection; Tier 4 updated; Post-fix row in Adversarial Review Scores table; `postfix_*` Pipeline Metrics fields; `rework_cycles` disambiguation |
| `.github/copilot-instructions.md` | Post-fix review bullet added to adversarial review protocol summary |
| `CLAUDE.md` | Step 5 Review description expanded with post-fix review |
| `Documents/Design/code-review.md` | Decision D21 added; pipeline diagram updated to show post-fix prosecution step |

## Validation Evidence

**Quick-validate (pre-PR):**
```
Plan-Architect references: 0  ✅
Janitor references:        0  ✅
Agent count:              13  ✅
```

**Per-step greps:**
```
Post-Fix Targeted Prosecution match in Code-Conductor.agent.md:  ≥1  ✅
post-fix targeted prosecution in Tier 4:                         ≥2  ✅
Post-fix Review row in scores table:                             ≥1  ✅
postfix_ fields in metrics YAML:                                 ≥5  ✅
Post-fix review in copilot-instructions.md:                       1  ✅
Post-fix review in CLAUDE.md:                                     1  ✅
D21 in code-review.md:                                            1  ✅
```

**Post-fix fix greps (PF1–PF7):**
```
postfix_judge_deferred present:                  2  ✅
postfix_findings_deferred gone:                  0  ✅
postfix_rework_cycles: n/a defined:              1  ✅
original implementation committed reasoning:     1  ✅
initial review cycle removed:                    0  ✅
explicit acceptance criterion item:              1  ✅
Evaluation ordering note:                        1  ✅
```

## CE Gate

⏭️ CE Gate not applicable — internal workflow documentation change; no customer-facing surface (`ce_gate: false` in plan)

Scenarios exercised: none (documentation-only, no runtime surface)

Track 2 (Process-Review): not applicable — no CE Gate ran

## Adversarial Review Scores

| Stage | Prosecutor | Defense | Judge rulings |
|-------|------------|---------|---------------|
| Code Review | 13 pts (13 sustained) | 1 pt (1 disproved, 1 rejected by judge) | 15 rulings |
| Post-fix Review | 7 pts (7 sustained) | 1 pt (1 disproved) | 8 rulings |
| CE Review | ⏭️ N/A | ⏭️ N/A | ⏭️ N/A |

## Process Gaps Found

None identified during this implementation cycle.

---

Closes #110

## Summary by Sourcery

Introduce a mandatory post-fix, diff-scoped adversarial review pass in the Code-Conductor workflow to re-prosecute specialist-applied fixes before CE Gate, and document its triggers, scope, and metrics across the review docs.

Enhancements:
- Add a documented post-fix targeted prosecution pass to the Code-Conductor agent workflow, including triggers based on severity and control-flow changes, recursion guard, and routing rules with a one-cycle loop budget.
- Extend adversarial review scoring and pipeline metrics documentation to track post-fix review execution and outcomes separately from the main review cycle.
- Update high-level review and pipeline design docs to describe the post-fix prosecution phase and capture it as a formal design decision.

### CODE REVIEW

File: .github/agents/Code-Conductor.agent.md [hunk 1]
```diff
@@ -246,6 +246,35 @@ For DEFERRED-SIGNIFICANT items, create a GitHub tracking issue automatically —
...
+
+**Evaluation ordering**: Condition 1 (severity) is evaluable immediately from the main-review judge output. Condition 2 (control flow) requires the fix diff and is evaluated after Diff scoping below — if condition 1 does not trigger alone, proceed through Tier 1 re-validation and Diff scoping before making the final skip decision.
+
+**Tier 1 re-validation** — After specialists apply fixes, re-run Tier 1 validation (build + lint/typecheck + tests). If Tier 1 fails, route the failure via the Failure Triage Rule and resolve it before proceeding.
+
+**Diff scoping** — After Tier 1 passes, compute the fix diff: `git diff HEAD -- {files touched by specialists}` isolates the review-fix changes (the original implementation was committed by the user before this phase; only specialist fix changes are uncommitted, so HEAD points to the pre-fix commit and the diff captures only review fixes). Pass those files and hunks in each prosecution prompt. Code-Critic runs in normal code prosecution mode (no marker) with the constrained input. Include the original PR change-type classification in each post-fix prosecution prompt (same requirement as main review pass prompts).
```
[the remark this answers is not in the document]
Comment (Grimblaz) [reply .github/agents/Code-Conductor.agent.md]: ✅ **ACCEPT** — Fixed in be1b2aa. Added an explicit **Diff scoping prerequisite** block that requires verifying git state (`git status`) before running the diff recipe. The original parenthetical explanation has been converted to a mandatory precondition check — if uncommitted implementation changes are present, the user must commit them first. The diff recipe itself now explicitly states what the constraint produces rather than explaining why it works.

> Proxy prosecution: 5 pts sustained | Defense: conceded | Judge: ✅ ACCEPT (AC item — diff-scoping correctness)
